File: python_backend/ranges/pawn/get_2_sqr_jump.py
from custom_except import *
from misc.step_funcs import step_1sqr270d, step_1sqr90d


def get_2_sqr_jump(sqr, color):
    """get the square 2 in front or 2 behind sqr"""
    if color == 'W':
        x, y = step_1sqr90d(sqr[0], sqr[1])
        x, y = step_1sqr90d(x, y)
    elif color == 'B':
        x, y = step_1sqr270d(sqr[0], sqr[1])
        x, y = step_1sqr270d(x, y)
    else:
        print('error: invalid color')
        return -1

    return (x, y)


if __name__ == "__main__":
    print(get_2_sqr_jump((1, 2), 'W'))
    print(get_2_sqr_jump((3, 7), 'B'))

File: app (1)/src/game_logic/ranges/pawn/get2SqrJump.js
import {step_1sqr90d, step_1sqr270d} from "../../helpers/stepFuncs";

export function get2SqrJump(sqr, color) {
    /*at this point it has been determined that there is a 2 square jump available so now get it**/
    var x, y;
    switch(color) {
        case 'W':
            [x, y] = step_1sqr90d(...sqr);
            [x, y] = step_1sqr90d(x, y);
            break;
        case 'B':
            [x, y] = step_1sqr270d(...sqr);
            [x, y] = step_1sqr270d(x, y);
            break;
        default:
            console.log('error: invalid color')
            [x, y] = [-1, -1]
            break;
    }

    return [x, y];
}

// module.exports = get2SqrJump;

/** for node.js
if (require.main === module) {
    console.log(get2SqrJump([1, 2], 'W'))
    console.log(get2SqrJump([3, 7], 'B'));

}
*/